site/static/api/device-registry-v1.yaml

```diff
@@ -0,0 +1,724 @@
+            204:
+               description: object deleted
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+# Device API
+
+   /devices/{tenantName}:
+
+      parameters:
+         - $ref: '#/components/parameters/tenantName'
+
+      post:
+         tags:
+            - devices
+         summary: Create new device registration with auto-generated ID
+         operationId: createDeviceRegistration
+         requestBody:
+            description: New device
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/Device'
+            required: true
+         responses:
+            201:
+               $ref: '#/components/responses/Created'
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+
+   /devices/{tenantName}/{deviceId}:
+
+      parameters:
+         - $ref: '#/components/parameters/tenantName'
+         - $ref: '#/components/parameters/deviceId'
+
+      post:
+         tags:
+            - devices
+         summary: Create new device registration
+         operationId: createDeviceRegistrationWithId
+         requestBody:
+            description: New device
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/Device'
+            required: true
+         responses:
+            201:
+               description: device registered
+               headers:
+                  Location:
+                     description: URL to the resource
+                     schema:
+                        type: string
+                        format: uri
+                  ETag:
+                     description: Version of the resource
+                     schema:
+                        type: string
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            409:
+               $ref: '#/components/responses/AlreadyExists'
+
+      get:
+         tags:
+            - devices
+         summary: Get device registration information
+         operationId: getRegistration
+         responses:
+            200:
+               description: operation successful
+               content:
+                  application/json:
+                     schema:
+                        $ref: '#/components/schemas/Device'
+               headers:
+                  ETag:
```

Comment:
this needs to be marked as *optional*, doesn't it?

Reply:
See above.